### PR TITLE
Fix for #2089

### PR DESCRIPTION
**Pull request by fgsch**
This addresses the 2 points from #2089.

### CODE REVIEW

File: lib/libvcc/vcc_expr.c [hunk 1]
```diff
@@ -1304,10 +1304,15 @@ vcc_expr0(struct vcc *tl, struct expr **e, vcc_type_t fmt)
...
-	if (fmt != (*e)->fmt && (fmt == STRING || fmt == STRING_LIST)) {
+	if (fmt == (*e)->fmt)
+		return;
+	if (fmt == STRING || fmt == STRING_LIST) {
 		vcc_expr_tostring(tl, e, fmt);
 		ERRCHK(tl);
-	}
+	} else if (fmt == DURATION && ((*e)->fmt == INT || (*e)->fmt == REAL))
+		(*e)->fmt = DURATION;
+	else if (fmt == REAL && ((*e)->fmt == DURATION || (*e)->fmt == INT))
```
Comment (nigoroll) [inline lib/libvcc/vcc_expr.c]: do we need to handle `fmt == INT` also?

File: lib/libvcc/vcc_expr.c [hunk 2]
```diff
@@ -146,13 +146,10 @@ vcc_DoubleVal(struct vcc *tl)
 void
 vcc_Duration(struct vcc *tl, double *d)
 {
-	double v, sc;
```
Comment (nigoroll) [inline lib/libvcc/vcc_expr.c]: Yes we should really drop the unit requirement, but accepting numbers as durations makes the syntax ambiguous. What if we kept the durations as-are and just folded `INT` into `DURATION`? IIUC this should be covered by your patch already and should achieve the same?

Comment (fgsch) [inline lib/libvcc/vcc_expr.c]: Not sure I follow. Mind sharing some VCL where this is visible?
All examples I can think of require dropping the unit, and in any case we should be consistent.

File: lib/libvcc/vcc_expr.c (hunk 1 above)
Comment (fgsch) [inline lib/libvcc/vcc_expr.c]: We could but first we need to define how we do the conversion from DURATION/REAL to INT (trunc or round). I'd rather handle this in a separate PR.

**Comment by fgsch** [comment]
#2089 is no more.

**Comment by hermunn** [comment]
Backport review: Nothing committed - nothing to backport.

**Pull request closed without merging**